src/util/apparmor.c

```diff
@@ -93,3 +163,489 @@ int bus_apparmor_dbus_supported(bool *supportedp) {
         *supportedp = supported;
         return 0;
 }
+
+static bool is_apparmor_enabled(void) {
+        int ret;
+        bool enabled;
+
+        ret = bus_apparmor_dbus_supported(&enabled);
+        if (ret || !enabled)
+                return false;
+
+        ret = bus_apparmor_is_enabled(&enabled);
+        if (ret || !enabled)
+                return false;
+
+        return true;
+}
+
+static int bus_apparmor_log(const char *fmt, ...) {
+        _c_cleanup_(c_freep) char *message = NULL;
+        va_list ap;
+        int r;
+
+        va_start(ap, fmt);
+        r = vasprintf(&message, fmt, ap);
+        va_end(ap);
+        if (r < 0)
+                return r;
+
+        /* XXX: we don't have access to any context, so can't find
+         * the right UID to use, follow dbus-daemon(1) and use our
+         * own. */
```

Comment:
Why can't we pass the context in? I don't understand this comment. Which context would be needed?

Reply:
I took over that function incl. the comment from the dbus-broker SELinux implementation without investigating further (and adapted it for apparmor).